appeals/web/src/server/lib/dates.js

```diff
@@ -1,196 +1,213 @@
-		date instanceof Date &&
-		!Number.isNaN(date) &&
-		!Number.isNaN(date.getDate()) &&
-		!Number.isNaN(date.getMonth()) &&
-		!Number.isNaN(date.getFullYear())
-	);
-};
+export const dateIsTodayOrInThePast = (dayMonthYearHourMinute) => {
+	const dateISOString = dayMonthYearHourMinuteToISOString(dayMonthYearHourMinute);
+	const todaysISOString = getTodaysISOString();
 
-/**
- * @param {import('../appeals/appeals.types.js').DayMonthYear} dayMonthYear
- * @returns {string} - date in format YYYY-MM-DD
- */
-export const dayMonthYearToApiDateString = (dayMonthYear) => {
-	const dayString = dayMonthYear.day < 10 ? `0${dayMonthYear.day}` : dayMonthYear.day;
-	const monthString = dayMonthYear.month < 10 ? `0${dayMonthYear.month}` : dayMonthYear.month;
+	if (dateISOString === todaysISOString) {
+		return true;
+	}
 
-	return `${dayMonthYear.year}-${monthString}-${dayString}`;
+	const date = new Date(dateISOString);
+	return isBefore(date, new Date(todaysISOString));
 };
 
 /**
- * @param {string|null|undefined} apiDateString - date in format YYYY-MM-DD
- * @returns {import('../appeals/appeals.types.js').DayMonthYear|undefined}
+ * @param {string | null | undefined} dateISOString
+ * @returns {string}
  */
-export const apiDateStringToDayMonthYear = (apiDateString) => {
-	if (apiDateString === null || apiDateString === undefined) {
-		return;
+export function dateISOStringToDisplayTime24hr(dateISOString) {
+	if (typeof dateISOString === 'undefined' || dateISOString === null) {
+		return '';
 	}
-	const date = parseISO(apiDateString);
 
-	if (!isValid(date)) {
-		logger.error('The date string provided parsed to an invalid date');
-		return;
+	let displayTimeString;
+
+	try {
+		displayTimeString = formatInTimeZone(dateISOString, timeZone, 'H:mm');
+	} catch (e) {
+		displayTimeString = '';
 	}
 
-	return dateToDayMonthYear(date);
-};
+	return displayTimeString;
+}
 
 /**
- * @param {string | undefined} hour
- * @param {string | undefined} minute
- * @returns {string} - time in format HH:MM
+ * @param {string | null | undefined} dateISOString
+ * @returns {string}
  */
-export const hourMinuteToApiDateString = (hour, minute = '0') => {
-	if (!hour) {
+export function dateISOStringToDisplayTime12hr(dateISOString) {
+	if (typeof dateISOString === 'undefined' || dateISOString === null) {
 		return '';
 	}
 
-	const hourString = hour.padStart(2, '0');
-	const minuteString = minute.padStart(2, '0');
+	let displayTimeString;
 
-	return `${hourString}:${minuteString}`;
-};
+	try {
+		displayTimeString = formatInTimeZone(dateISOString, timeZone, `h:mmaaaaa'm'`);
+	} catch (e) {
+		displayTimeString = '';
+	}
+
+	return displayTimeString;
+}
 
 /**
- * @param {Date | number | string | null | undefined} date
- * @param {{ condensed?: boolean }} options
  * @returns {string}
  */
-export function dateToDisplayDate(date, { condensed = false } = {}) {
-	if (typeof date === 'undefined' || date === null) {
-		return '';
-	}
+export function getTodaysISOString() {
```

Comment:
This appears to be specifically "midnight today"? Seems fine, but perhaps the jsdoc comment could mention that.

Reply:
done